riscv/processor.h

```diff
@@ -252,17 +252,25 @@ class processor_t : public abstract_device_t
   mmu_t* get_mmu() { return mmu; }
   state_t* get_state() { return &state; }
   unsigned get_xlen() const { return xlen; }
```

Comment:
'xlen' is for current privilege mode. And it's updated if privilege mode changes. It's used by every instruction function, so I think it can get better performance to keep it alone instead of getting it from CSRs every time. Maybe the other *xlen's are not very necessary.